src/transformers/utils/fx.py

```diff
@@ -616,6 +671,20 @@ def _proxies_to_metas(v):
     return v
 
 
+def _patch_fn_for_proxy(fn, tracer):
+    def _apply_on_proxy(x):
+        if isinstance(x, torch.fx.Proxy):
+            if not (isinstance(x, HFProxy) and hasattr(x, "_metadata")):
```

Comment:
Maybe we can test that at the top of `_apply_on_proxy` and remove the first test.

Reply:
Discussed offline, there's three outcomes, so necessarily you need at least two if statements.